From 7c9b02af520a62894fab39c606e0c3a321e6b0b0 Mon Sep 17 00:00:00 2001
From: Julien Desfossez <julien.desfossez@polymtl.ca>
Date: Fri, 4 Feb 2011 11:08:38 -0500
Subject: [PATCH 228/248] net_dev_xmit_extended probe : cleanup unused data

Intialise the unused fields of the data structure and add support for
ports in UDP packets.

Signed-off-by: Julien Desfossez <julien.desfossez@polymtl.ca>
Signed-off-by: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
---
 ltt/probes/net-extended-trace.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/ltt/probes/net-extended-trace.c b/ltt/probes/net-extended-trace.c
index 15fc810..d7bb4f8 100644
--- a/ltt/probes/net-extended-trace.c
+++ b/ltt/probes/net-extended-trace.c
@@ -42,6 +42,9 @@ notrace void probe_net_dev_xmit_extended(void *_data, struct sk_buff *skb)
 	struct serialize_l214421224411111 data;
 	struct iphdr *iph = ip_hdr(skb);
 	struct tcphdr *th = tcp_hdr(skb);
+	struct udphdr *uh = udp_hdr(skb);
+
+	memset(&data, 0, sizeof(struct serialize_l214421224411111));
 
 	data.f1 = (unsigned long)skb;
 	data.f2 = skb->protocol;
@@ -63,6 +66,9 @@ notrace void probe_net_dev_xmit_extended(void *_data, struct sk_buff *skb)
 			data.f14 = th->rst;
 			data.f15 = th->syn;
 			data.f16 = th->fin;
+		} else if (data.f3 == IPPROTO_UDP) {
+			data.f8 = uh->source;
+			data.f9 = uh->dest;
 		}
 	}
 
-- 
1.7.0.4

